crates/composition/src/compose_supergraph.rs

```diff
@@ -70,6 +75,10 @@ fn merge_object_definitions<'a>(
 fn merge_field_definitions(ctx: &mut Context<'_>, fields: &[FieldWalker<'_>]) {
     let Some(first) = fields.get(0) else { return };
 
+    if first.parent_definition().kind() != DefinitionKind::Object {
+        return;
+    }
```

Comment:
Not done in this PR, but I think the way this is going to evolve is matching on the definition kind here too. Then we can specialize on merging interface fields, object fields or input object fields. In separate functions of course for readability.